src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1102,12 +1102,62 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
 
                 // Don't lower if we're not emitting or if there were errors. 
                 // Methods that had binding errors are considered too broken to be lowered reliably.
-                if (_moduleBeingBuiltOpt == null || hasErrors)
+                if (_moduleBeingBuiltOpt == null || hasErrors || isEmptySynthesizedStaticConstructor(processedInitializers.BoundInitializers))
                 {
                     _diagnostics.AddRange(actualDiagnostics);
                     return;
                 }
 
+                // We should only go through with emitting a synthesized static constructor if it would contain
+                // "meaningful" statements. If it only initializes fields to default values we will skip emitting it.
+                bool isEmptySynthesizedStaticConstructor(ImmutableArray<BoundInitializer> initializers)
+                {
+                    if (methodSymbol is SynthesizedStaticConstructor)
+                    {
+                        foreach (var initializer in initializers)
+                        {
+                            var value = (initializer as BoundFieldEqualsValue)?.Value;
+                            if (value is null)
+                            {
+                                // this isn't a BoundFieldEqualsValue, so this initializer is doing
+                                // something we don't understand. Better just emit it.
+                                return false;
+                            }
+
+                            // we are assigning 'default(SomeType)' to something.
+                            if (value is BoundDefaultExpression)
+                            {
+                                continue;
+                            }
```

Comment:
If the type is a value type, shouldn't we emit the .cctor?
```
private static S _s = default;
```

Reply:
We expect the runtime to zero out such fields. The optimization that removes redundant initializers will remove this initializer. See the empty .cctor in this example: https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8ACAmARgFgAoHAZgAJYEBXAYxkoGUDKBvUy7yiytAHZMAZmjgYAJgG5SAX1Kk++SgGEOXHkoIA2FmyhsAvJQlxhEWhhgySsoA==